src/NServiceBus.RabbitMQ/RabbitMqDequeueStrategy.cs

```diff
@@ -280,7 +279,26 @@ void Purge()
             }
         }
 
+        bool MovePoisonMessage(BasicDeliverEventArgs message, string queue)
+        {
+            var connection = connectionManager.GetPublishConnection();
+            var success = true;
```

Comment:
Perhaps it would be safer to set this to false, and only set it to true as the last statement in the `try` block? Or we could even remove it and do `return true` and `return false` in the appropriate places.